chocolatey/plugins/module_utils/Common.psm1

```diff
@@ -0,0 +1,184 @@
+    [CmdletBinding()]
+    param()
+
+    $script:module
+}
+
+function Get-ChocolateyCommand {
+    <#
+        .SYNOPSIS
+        Retrieves a CommandInfo object for `choco.exe` if it is present on the system.
+
+        .DESCRIPTION
+        Returns either a CommandInfo object which contains the path the `choco.exe`
+        or registers a task failure and exits if it cannot be found.
+    #>
+    [CmdletBinding()]
+    param(
+        # If provided, does not terminate the task run when choco.exe is not found.
+        [Parameter()]
+        [switch]
+        $IgnoreMissing
+    )
+
+    $command = Get-Command -Name choco.exe -CommandType Application -ErrorAction SilentlyContinue
+
+    if (-not $command) {
+        $installDir = if ($env:ChocolateyInstall) {
+            $env:ChocolateyInstall
+        }
+        else {
+            "$env:SYSTEMDRIVE\ProgramData\Chocolatey"
+        }
+
+        $command = Get-Command -Name "$installDir\bin\choco.exe" -CommandType Application -ErrorAction SilentlyContinue
+
+        if (-not ($command -or $IgnoreMissing)) {
+            $message = "Failed to find Chocolatey installation, make sure choco.exe is in the PATH env value"
+            Assert-TaskFailed -Message $message
+        }
+    }
+
+    $command
+}
+
+function Assert-TaskFailed {
+    <#
+        .SYNOPSIS
+        Notifies Ansible that the task failed, and exits after providing the necessary details.
+
+        .DESCRIPTION
+        Exits the current Ansible task with a failure state, recording a message as well as
+        any other relevant information for the current task.
+    #>
+    [CmdletBinding()]
+    param(
+        # Message to be returned to the task runner, indicating why the failure happened.
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Message,
+
+        # The Ansible module object to register the failure with.
+        # Defaults to the currently active module.
+        [Parameter()]
+        [Ansible.Basic.AnsibleModule]
+        $Module = (Get-AnsibleModule),
+
+        # The native command call that resulted in the error, if any.
+        [Parameter()]
+        [string]
+        $Command,
+
+        # A hashtable containing `stdout`, `stderr`, and `rc` keys resulting from
+        # a native command execution, which will be copied to the module's Result
+        # dictionary before returning the fail state.
+        [Parameter()]
+        [hashtable]
+        $CommandResult,
+
+        # An exception to provide to the `$module.ExitJson()` method with further
+        # information about the error, if any.
+        [Parameter()]
+        [Exception]
+        $Exception
+    )
+
+    if ($null -ne $CommandResult) {
+        $resultKeys = 'rc', 'stdout', 'stderr'
+
+        foreach ($key in $resultKeys) {
+            $Module.Result.$key = $CommandResult.$key
+        }
+    }
+
+    if ($null -ne $Command) {
+        $Module.Result.command = $Command
+    }
+
+    if ($null -ne $Exception) {
```

Comment:
Aye, I mostly moved things into this helper command so we could avoid writing up the failure handling multiple times, since we were doing pretty similar stuff every time.

I guess it's not strictly correct/common to use the `Assert` verb for a command which will always call `exit`, but it was the closest I could come up with. 😁